src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -1,420 +1,505 @@
-        /// Handle request to pause a profiling session
-        /// </summary>
-        internal async Task HandleGetXEventSessionsRequest(GetXEventSessionsParams parameters, RequestContext<GetXEventSessionsResult> requestContext)
-        {
-            var result = new GetXEventSessionsResult();
-            ConnectionInfo connInfo;
-            ConnectionServiceInstance.TryFindConnection(
-                parameters.OwnerUri,
-                out connInfo);
-            if (connInfo == null)
-            {
-                await requestContext.SendError(new Exception(SR.ProfilerConnectionNotFound));
-            }
-            else
-            {
-                List<string> sessions = GetXEventSessionList(parameters.OwnerUri, connInfo);
-                result.Sessions = sessions;
-                await requestContext.SendResult(result);
-            }
-        }
-
-        /// <summary>
-        /// Handle request to disconnect a session
-        /// </summary>
-        internal async Task HandleDisconnectSessionRequest(DisconnectSessionParams parameters, RequestContext<DisconnectSessionResult> requestContext)
+                catch (Exception e)
+                {
+                    await requestContext.SendError(new ProfilerException(SR.StartSessionFailed(e.Message), e));
+                }
+            });
+        }
+
+        private async Task StartLocalFileSession(StartProfilingParams parameters, RequestContext<StartProfilingResult> requestContext)
         {
-            monitor.StopMonitoringSession(parameters.OwnerUri, out _);
+            var xeSession = XEventSessionFactory.OpenLocalFileSession(parameters.SessionName);
+            monitor.StartMonitoringSession(parameters.OwnerUri, xeSession);
+            xeSession.Start();
+            var result = new StartProfilingResult() { UniqueSessionId = xeSession.Id.ToString(), CanPause = false };
+            await requestContext.SendResult(result);
         }
 
-        /// <summary>
-        /// Gets a list of all running XEvent Sessions
-        /// </summary>
-        /// <returns>
-        /// A list of the names of all running XEvent sessions
-        /// </returns>
-        internal List<string> GetXEventSessionList(string ownerUri, ConnectionInfo connInfo)
-        {
-            var sqlConnection = ConnectionService.OpenSqlConnection(connInfo);
-            SqlStoreConnection connection = new SqlStoreConnection(sqlConnection);
-            BaseXEStore store = CreateXEventStore(connInfo, connection);
-
-            // get session names from the session list
-            List<string> results = store.Sessions.Aggregate(new List<string>(), (result, next) =>
+        /// <summary>
+        /// Handle request to stop a profiling session
+        /// </summary>
+        internal async Task HandleStopProfilingRequest(StopProfilingParams parameters, RequestContext<StopProfilingResult> requestContext)
+        {
+            await Task.Run(async () =>
             {
```

Comment:
Pls remind me why these tasks need to use Task.Run explicitly. Why is the use of `await` in the body of the method insufficient?

Reply:
This is merge mess up. Working on it.

Reply:
Fixed this in next iteration.